sensor/common/detector/networkbaseline/network_baseline.go

```diff
@@ -70,7 +71,24 @@ func (e *networkBaselineEvaluator) checkPeerInBaselineForEntity(
 		// If no baseline exists then we do not mark it as anomalous
 		return true
 	}
-	peer := networkbaseline.PeerFromNetworkEntityInfo(peerEntity, peerEntityName, dstPort, protocol, isIngressToBaselineEntity)
+
+	var peer networkbaseline.Peer
+	if peerEntity.GetType() == storage.NetworkEntityInfo_EXTERNAL_SOURCE &&
+		peerEntity.GetExternalSource().GetDiscovered() {
```

Comment:
Seeing `GetDiscovered()` returning a bool feels less natural than for `IsDiscovered`. No need to rename that here, but maybe we should do that on the next occasion.

Reply:
Yeah I agree it's an unfortunate one, but it's a feature of the protobuf code generation and not something we can change (as far as I understand, at least)